### PR TITLE
more consolidated mesh executables and cleanup 

### PR DESCRIPTION
**Pull request by cchriste**
This is new work consolidating various mesh executables. Not sure if any of it needs to go into 6.0, but from here on out we're going to try to keep our branches more specifically named and incremental.

Changes:
- removing default spacing for Image::setSpacing function *
- addition of handy Image::rotate function that simply accepts Axis::X/Y/Z, in addition to more general version *
- removal of GenericMeshConversion util (including VTKUnstructuredGridToPolyData, FeaFromVTK, vtk2ply, stl2vtk, stl2ply, ply2vtk)
- addition of Mesh::clipClosedSurface function *
- removal of unnecessary RasterizationOrigin and RasterizationSize commands (private functions in Mesh)
- fixes and simplifications for clipping in GroomUtils
- misc cleanups and clarifications in Image, Commands, GroomUtils, and Python bindings
- removal of ExtractVertexFeatureFromMesh 
- removal of WriteFeatureGradientAsText
- added notes in code for related GitHub issues (e.g., adding ability to use google test for private functions)

*and associated command line and python bindings

### CODE REVIEW

**Comment by cchriste** [comment]
TODO before this is ready:

**For Mesh::clipClosedSurface, add...**
- [ ]  MeshTest
- [ ]  shapeworksTest
- [ ]  python binding
- [ ]  pythonTest

**Review by archanasri — APPROVED** [review]

**Review by akenmorris — CHANGES_REQUESTED** [review]
This looks like a lot of great additions!

I'm noticing a lot of inconsistencies in command names with regard to word separation.  For example, I see "readparticlesystem" in some parts of the repo and "read-particle-system" in other parts.

**Comment by archanasri** [comment]
@akenmorris Our command line interface supports both: read-particle-system and readparticlesystem. All the test cases use without hyphens.

File: Applications/shapeworks/MeshCommands.cpp [hunk 1]
```diff
@@ -153,17 +153,17 @@ bool Coverage::execute(const optparse::Values &options, SharedCommandData &share
   }
 
   const std::string otherMesh(static_cast<std::string>(options.get("name")));
-  bool allowBackIntersections = static_cast<bool>(options.get("allowBackIntersections"));
-  double angleThreshold = static_cast<double>(options.get("angleThreshold"));
-  double backSearchRadius = static_cast<double>(options.get("backSearchRadius"));
+  bool allowbackintersections = static_cast<bool>(options.get("allowbackintersections"));
```
[the remark this answers is not in the document]
Comment (cchriste) [reply Applications/shapeworks/MeshCommands.cpp]: I updated all the variables in the commands to use camelCase, keeping the arguments themselves all lowercase.

**Comment by cchriste** [comment]
@akenmorris if you feel this is complete please go ahead and merge and delete the branch. Let us party no more.

**Review by akenmorris — APPROVED** [review]

**Comment by akenmorris** [comment]
Thanks for updating the C++ variable names.

Regarding the command line interface, I'll just comment that I disagree with multi-word options using all lowercase and no word separation (e.g. --allowbackintersections).  That one is probably fairly readable, but I can anticipate problems with this approach in the future.  If this is ok with you, @sheryjoe , then let's go ahead and I will keep quiet.  I just thought we had decided to use dashes at a meeting a while back.  Was this decision reversed and I just missed it?

I'm just looking at other command line interfaces I commonly use (e.g. c3d, ImageMagick, `man ls`, `git help`, `gcc --help`), and they all use dashes.  I'm not seeing examples of multi-word options with no word separation.  Can I ask what the rational is for this approach?

**Comment by sheryjoe** [comment]
> Regarding the command line interface, I'll just comment that I disagree with multi-word options using all lowercase and no word separation (e.g. --allowbackintersections). 

I agree with @akenmorris. We have discussed this before in the context of subcommand names when we started the consolidation work to make sure that subcommands are readable without having to remember CamelCased composite words. The consensus was that we will use multi-word commands with dashes/hyphens for word separation with the option of no-dashes for developer(s) convenience.  Commands arguments/options are no different here.

**Comment by cchriste** [comment]
> I agree with @akenmorris. We have discussed this before in the context of subcommand names when we started the consolidation work to make sure that subcommands are readable without having to remember CamelCased composite words. The consensus was that we will use multi-word commands with dashes/hyphens for word separation with the option of no-dashes for developer(s) convenience. Commands arguments/options are no different here.

Great! I just created #1056 to change all of them. If possible, let's merge this PR and address that separately since it's a broader issue requiring updating of all examples as well as the commands themselves.

**Pull request merged**